### PR TITLE
Add store REST API for SDK generation.

### PR DESCRIPTION
**Pull request by chamindias**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -449,4 +454,65 @@ public Response apisApiIdThumbnailGet(String apiId, String accept, String ifNone
...
+        //this condition is to evaluate if the supported language list is non empty
+        if (StringUtils.isNotBlank(supportedSDKLanguages)) {
+            //this boolean checks if the required language is in the supported language list for SDK generation
+            boolean isLanguageSupported = Arrays.asList(supportedSDKLanguages.split(",")).contains(language);
+            if (!isLanguageSupported) {
+                String errorMessage = "SDK generation is not supported for language : " + language;
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+        } else {
+            String errorMessage = "No supported languages for SDK generation.";
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java]: No use of having the error message in a separate variable.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java [hunk 2]
```diff
@@ -449,4 +454,65 @@ public Response apisApiIdThumbnailGet(String apiId, String accept, String ifNone
...
+        Map<String, String> sdkDataMap = null;
+        try {
+            sdkDataMap = apiClientGenerationManager.generateSDK(language, apiIdentifier.getApiName(),
+                    apiIdentifier.getVersion(), apiIdentifier.getProviderName());
+        } catch (APIClientGenerationException e) {
+            String errorMessage = "SDK generation failed. Error : " + e.getMessage();
+            RestApiUtil.handleInternalServerError(errorMessage, log);
+        }
+        if (sdkDataMap != null) {
+            File file = new File(sdkDataMap.get("zipFilePath"));
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java]: SHouldn't zipFilePath be a constant?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -449,4 +454,65 @@ public Response apisApiIdThumbnailGet(String apiId, String accept, String ifNone
...
+
+        if (StringUtils.isBlank(apiId) || StringUtils.isBlank(language)) {
+            String errorMessage = "API ID or SDK language should not be empty or null.";
+            RestApiUtil.handleBadRequest(errorMessage, log);
+        }
+        APIIdentifier apiIdentifier = null;
+        try {
+            apiIdentifier = APIMappingUtil.getAPIIdentifierFromApiIdOrUUID(apiId, xWSO2Tenant);
+        } catch (APIManagementException e) {
+            RestApiUtil.handleResourceNotFoundInTenantError(RestApiConstants.RESOURCE_API, apiId, log, xWSO2Tenant);
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApisApiServiceImpl.java]: I think its better to log the exception(e) here as well.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java [hunk 4]
```diff
@@ -323,6 +323,25 @@ public static NotFoundException buildNotFoundException(String resource, String i
     /**
      * Returns a new NotFoundException
      *
+     * @param resource resource type
+     * @param id       identifier of the resource
+     * @param tenant   tenant for which the resource has been searched
+     * @return a new NotFoundException with the specified details
+     */
+    public static NotFoundException buildNotFoundException(String resource, String id, String tenant) {
```
Comment (indikasampath2000) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java]: Method name should start with Capital letter.

**Comment by indikasampath2000** [comment]
Yeah.. Sorry.. My bad..

**Pull request merged**